### PR TITLE
Leaderboard hit the API

### PR DESCRIPTION
**Pull request by Bria222**
### Implimentented the following features
- Sending and receiving data from an API.
- Using API documentation.
-  formatted data into JSON. format
- Made my JavaScript code asynchronous.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -1,18 +1,34 @@
...
+  await postApi(inputFields);
   inputNameField.value = '';
   inputScoreField.value = '';
+});
+
+refreshButton.addEventListener('click', () => {
+  scoreContainer.innerHTML = '';
+  fetch('https://us-central1-js-capstone-backend.cloudfunctions.net/api/games/llFawVoYD7zLAfvY9jQn/scores/')
+    .then((res) => res.json())
+    .then((data) => {
```
Comment (AnselemOdims) [inline src/index.js]: - [x] Please kindly use `async` and `await` for your asynchronous functions as required.

File: src/fetchApi.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+const postApi = (score) => {
+  const scoreResult = fetch(apiUrl, {
+    method: 'POST',
+    headers: {
+      'Content-Type': 'application/json; charset=UTF-8',
+    },
+    body: JSON.stringify(score),
+  });
+  return scoreResult.json;
```
Comment (AnselemOdims) [inline src/fetchApi.js]: - [x] As pointed out earlier, please use `async` and `await`

File: src/index.js [hunk 3]
```diff
@@ -1,18 +1,34 @@
...
+  template.innerHTML = `<p><span>${inputFields.user}: </span> ${inputFields.score}</p>`;
   scoreContainer.appendChild(template.content);
+  await postApi(inputFields);
   inputNameField.value = '';
   inputScoreField.value = '';
+});
+
+refreshButton.addEventListener('click', () => {
+  scoreContainer.innerHTML = '';
+  fetch('https://us-central1-js-capstone-backend.cloudfunctions.net/api/games/llFawVoYD7zLAfvY9jQn/scores/')
```
Comment (AnselemOdims) [inline src/index.js]: - [x] Please kindly make sure the items are getting fetched when the browser is reloaded.

**Review by AnselemOdims — CHANGES_REQUESTED** [review]
### Hi @Bria222 , 
You have been doing a tremendous job so far :ok_hand:
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight 🎉
PR has a title and summary. :heavy_check_mark:
No linter errors :heavy_check_mark:

## Required Changes ♻️
- [ ] Please also check inline comments under the review

## Optional suggestions
_Comments with the [OPTIONAL] prefix are not a must to implement however, It's strongly recommended to work on them as they can make your code better._

## Happy coding!👏👏👏

#### Please leave any questions or comments in the PR if you have any doubts or unclarity! You can tag me using @AnselemOdims for a quicker response

**_Note: Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: src/index.js (hunk 1 above)
Comment (Bria222) [reply src/index.js]: okay let me rewrite it

File: src/fetchApi.js (hunk 2 above)
Comment (Bria222) [reply src/fetchApi.js]: @AnselemOdims okay let me rewrite the logic

File: src/index.js (hunk 3 above)
Comment (Bria222) [reply src/index.js]: @AnselemOdims i have fixed that

**Review by arslanbisharat — APPROVED** [review]
Hi,
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights

- The Pull Request has a proper title and description. ✔️ 
- You have added a descriptive Readme file. Good Job. ✔️ 
- There is no linter error is present. ✔️ 
- Project has met all the technical aspects you needed for this project. Keep rocking! ✔️ 


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**